### ISSUE TITLE
Future snapshot is not the route navigated to

### ISSUE THREAD

**Issue by chrillewoodz** [angular/angular#22116]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Feature request
[x] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior
I'm not sure whether this is a bug or just a documentation error but for `shouldReuseRoute`, the `future` parameter is not the route you're navigating to which is really confusing. 

## Expected behavior
I would expect the `future` parameter to be the route that you're navigating to.

## Minimal reproduction of the problem with instructions
Create two routes, and go from one to the other.

## What is the motivation / use case for changing the behavior?
The current behaviour makes it hard to work with the `RouteReuseStrategy`.

## Environment

<pre><code>
Angular version: 5.2.1

</code></pre>

**Comment by mhevery** [comment]
Hello, we reviewed this issue and determined that it doesn't fall into the bug report or feature request category. This issue tracker is not suitable for support requests, please repost your issue on [StackOverflow](http://stackoverflow.com/) using tag `angular`.

If you are wondering why we don't resolve support issues via the issue tracker, please [check out this explanation](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-got-a-question-or-problem).

**Comment by chrillewoodz** [comment]
@mhevery How is it a support request? I'm not asking for help, I'm reporting a flaw in your framework.

**Comment by riheldo** [comment]
@chrillewoodz I realized the same problem. I believed the names of the variables are changed. So decided to make some testes, *results below*. 

I'm moving from **/dashboad/products** to **/dashboard/ticket/3**

<img width="835" alt="captura de tela 2018-07-30 as 21 14 44" src="https://user-images.githubusercontent.com/4129170/43430211-2990cb22-943e-11e8-8289-9192d910f1f4.png">

The shouldReuseRoute function is called at least 3 times and I noticed that the behavior of the variables change between calls. Which leads me to believe that calling them current and future are not appropriate names.

So, @mhevery , i think its a documentation problem.
